src/plugins/explore/public/components/query_panel/components/editor_stack/prompt_editor.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { EuiIcon } from '@elastic/eui';
+import { getEditorConfig, LanguageType } from './shared';
+import { ReusableEditor } from './resuable_editor';
+
+export interface PromptEditorProps {
+  languageType: LanguageType;
+  prompt: string;
+  isPromptReadOnly: boolean;
+  onChange: (value: string) => void;
+  onPromptRun: (queryString?: string) => void;
+  onPromptEdit: () => void;
+  onClearEditor: () => void;
+}
+
+export const PromptEditor: React.FC<PromptEditorProps> = ({
+  languageType,
+  prompt,
+  isPromptReadOnly,
+  onChange,
+  onPromptRun,
+  onPromptEdit,
+  onClearEditor,
+}) => {
+  const editorConfig = getEditorConfig(languageType);
+  const langText = languageType !== 'nl' ? 'query' : 'prompt';
+  return (
+    <ReusableEditor
+      value={prompt}
+      onChange={onChange}
+      onRun={onPromptRun}
+      isReadOnly={isPromptReadOnly}
+      onEdit={onPromptEdit}
+      onClear={onClearEditor}
+      editorConfig={editorConfig}
+      placeholder={
+        <>
+          Ask a question or search using <EuiIcon type="editorCodeBlock" /> PPL
```

Comment:
1. I18n
2. PPL should not be hardcoded here. We should read the selected language from somewhere

Reply:
can we write a TODO comment?